src/core/transaction/database.pl

```diff
@@ -275,8 +275,8 @@
     commit_validation_objects(Validations, Committed),
     collect_validations_metadata(Validations, Validation_Meta_Data),
     collect_commit_metadata(Committed, Commit_Meta_Data),
-    put_dict(Validation_Meta_Data, Commit_Meta_Data, Meta_Data).
-
+    put_dict(Validation_Meta_Data, Commit_Meta_Data, Meta_Data),
+    ignore(foreach(post_commit_hook(Validations, Meta_Data), true)).
```

Comment:
Not sure if this needs to be `foreach` rather than `forall`. What things are going to become more bound given that Validations and Meta_Data are already bound?

Reply:
You're right, it should just all be forall.

Reply:
or it could be `exhaust` from utils.pl